utilities/transactions/lock/point/point_lock_manager.cc

```diff
@@ -32,34 +127,133 @@ struct LockInfo {
       : exclusive(ex), expiration_time(time) {
     txn_ids.push_back(id);
   }
-  LockInfo(const LockInfo& lock_info)
 
-      = default;
-  void operator=(const LockInfo& lock_info) {
-    exclusive = lock_info.exclusive;
-    txn_ids = lock_info.txn_ids;
-    expiration_time = lock_info.expiration_time;
-  }
-  DECLARE_DEFAULT_MOVES(LockInfo);
+  // waiter queue for this key
+  std::unique_ptr<std::list<std::unique_ptr<KeyLockWaiter>>> waiter_queue;
 };
 
 struct LockMapStripe {
   explicit LockMapStripe(std::shared_ptr<TransactionDBMutexFactory> factory) {
     stripe_mutex = factory->AllocateMutex();
-    stripe_cv = factory->AllocateCondVar();
     assert(stripe_mutex);
-    assert(stripe_cv);
+
+    mutex_factory_ = std::move(factory);
+    FillKeyCVPool();
+  }
+
+  // Allocate a batch of mutex and conditional variable pairs to minimize memory
+  // fragmentation
+  void FillKeyCVPool() {
+    // get number of CPU count
```

Comment:
I agree with a pool per stripe, but this seems like a lot of entries to allocate per stripe, as the number of stripes should be on the order of number of CPU cores.